web/src/beta/components/fields/SpacingInput/index.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from "react";
+
+import { styled } from "@reearth/services/theme";
+
+import Property from "..";
+import NumberInput from "../common/NumberInput";
+
+export type SpacingValues = {
+  top: number;
+  left: number;
+  right: number;
+  bottom: number;
+};
+
+type Props = {
+  name?: string;
+  description?: string;
+  value?: SpacingValues;
+  min?: number;
+  max?: number;
+  onChange?: (values: SpacingValues) => void;
+};
+
+const SpacingInput: React.FC<Props> = ({ name, description, value, min, max, onChange }) => {
+  const [spacingValues, setSpacingValues] = useState<SpacingValues>(
+    value || { top: 0, left: 0, right: 0, bottom: 0 },
+  );
+
+  const handleInputChange = (
+    position: keyof SpacingValues,
+    newValue: string | number | undefined,
+  ) => {
+    const updatedValues = { ...spacingValues, [position]: newValue };
+    setSpacingValues(updatedValues);
+    onChange?.(updatedValues);
+  };
+
+  return (
+    <Property name={name} description={description}>
+      <StyledRectangle>
+        {["top", "left", "right", "bottom"].map(position => (
+          <SpacingField
+            value={getSpacingPosition(spacingValues, position as keyof SpacingValues)}
```

Comment:
Can you move the GetSpacingPo.... code in to a useMemo and use that instead?

Reply:
done